gen2-palm-detection/main.py

```diff
@@ -0,0 +1,468 @@
+    h = raw_boxes[..., 3] / y_scale * anchors[:, 3]
+
+    boxes[..., 1] = y_center - h / 2.0  # xmin
+    boxes[..., 0] = x_center - w / 2.0  # ymin
+    boxes[..., 3] = y_center + h / 2.0  # xmax
+    boxes[..., 2] = x_center + w / 2.0  # ymax
+
+    for k in range(num_keypoints):
+        offset = 4 + k * 2
+        keypoint_x = raw_boxes[..., offset] / x_scale * anchors[:, 2] + anchors[:, 0]
+        keypoint_y = (
+            raw_boxes[..., offset + 1] / y_scale * anchors[:, 3] + anchors[:, 1]
+        )
+        boxes[..., offset] = keypoint_x
+        boxes[..., offset + 1] = keypoint_y
+
+    return boxes
+
+
+def raw_to_detections(raw_box_tensor, raw_score_tensor, anchors_, shape, num_keypoints):
+    """
+
+    This function converts these two "raw" tensors into proper detections.
+    Returns a list of (num_detections, 17) tensors, one for each image in
+    the batch.
+
+    This is based on the source code from:
+    mediapipe/calculators/tflite/tflite_tensors_to_detections_calculator.cc
+    mediapipe/calculators/tflite/tflite_tensors_to_detections_calculator.proto
+    """
+    detection_boxes = decode_boxes(raw_box_tensor, anchors_, shape, num_keypoints)
+    detection_scores = sigmoid(raw_score_tensor).squeeze(-1)
+    output_detections = []
+    for i in range(raw_box_tensor.shape[0]):
+        boxes = detection_boxes[i]
+        scores = np.expand_dims(detection_scores[i], -1)
+        output_detections.append(np.concatenate((boxes, scores), -1))
+    return output_detections
+
+
+def non_max_suppression(boxes, probs=None, angles=None, overlapThresh=0.3):
+    if len(boxes) == 0:
+        return [], []
+
+    if boxes.dtype.kind == "i":
+        boxes = boxes.astype("float")
+
+    pick = []
+
+    x1 = boxes[:, 0]
+    y1 = boxes[:, 1]
+    x2 = boxes[:, 2]
+    y2 = boxes[:, 3]
+
+    area = (x2 - x1 + 1) * (y2 - y1 + 1)
+    idxs = y2
+
+    if probs is not None:
+        idxs = probs
+
+    idxs = np.argsort(idxs)
+
+    while len(idxs) > 0:
+        last = len(idxs) - 1
+        i = idxs[last]
+        pick.append(i)
+
+        xx1 = np.maximum(x1[i], x1[idxs[:last]])
+        yy1 = np.maximum(y1[i], y1[idxs[:last]])
+        xx2 = np.minimum(x2[i], x2[idxs[:last]])
+        yy2 = np.minimum(y2[i], y2[idxs[:last]])
+
+        w = np.maximum(0, xx2 - xx1 + 1)
+        h = np.maximum(0, yy2 - yy1 + 1)
+
+        overlap = (w * h) / area[idxs[:last]]
+
+        idxs = np.delete(
+            idxs, np.concatenate(([last], np.where(overlap > overlapThresh)[0]))
+        )
+
+    if angles is not None:
+        return boxes[pick].astype("int"), angles[pick]
+    return boxes[pick].astype("int")
+
+
+class Main(DepthAI):
+    def __init__(self, file=None, camera=False):
+        self.cam_size = (128, 128)
+        super().__init__(file, camera)
+
+    def create_nns(self):
+        self.create_nn(
+            "models/palm_detection_openvino_2020.1_4shave.blob", "palm", first=True
```

Comment:
Can you compile the model to the latest 2021.3 OpenVINO? We have recently released a [blobconverter](https://pypi.org/project/blobconverter/) tool that can help you in compiling it faster